nilearn/datasets/data/MAIN_osf.csv

```diff
@@ -1,84 +1,84 @@
-hash_r,participant_id,hash_b
-uhd7y,sub-pixar001,2vck8
-rx2ae,sub-pixar003,vahys
-zgjb5,sub-pixar004,kw6b2
-2eyhs,sub-pixar005,jzqmr
-bmpcz,sub-pixar006,r729e
-38apf,sub-pixar007,p9ckg
-gb4ze,sub-pixar009,yja4w
-ym63x,sub-pixar010,3w9b8
-svxmr,sub-pixar011,ybp2x
-s3hp6,sub-pixar012,8q9my
-uftvz,sub-pixar015,geunp
-s5kcq,sub-pixar018,r6f37
-sd5m3,sub-pixar020,m8vaf
-c5gwk,sub-pixar022,hn7m4
-b5w72,sub-pixar027,c68mx
-rh6sc,sub-pixar029,s5vmg
-h3v2n,sub-pixar030,6g4en
-qz5kr,sub-pixar031,wp6yx
-adwcg,sub-pixar035,67hbq
-6gjbw,sub-pixar039,2tqhw
-gxjds,sub-pixar041,nqes4
-gackh,sub-pixar042,xtay5
-jy63r,sub-pixar044,5kqdz
-6z9je,sub-pixar046,jxe3s
-9unqp,sub-pixar049,6gazy
-469av,sub-pixar050,6bktv
-n4k63,sub-pixar052,gcte3
-7baex,sub-pixar057,28u3w
-c3swg,sub-pixar058,x3s75
-gdj6y,sub-pixar061,sx38f
-wesjp,sub-pixar062,xsb95
-kqn5y,sub-pixar063,pc27u
-3bfcp,sub-pixar065,d5ura
-7r2df,sub-pixar067,7xrfb
-3tb27,sub-pixar068,q2rjg
-t8m4r,sub-pixar072,23h9j
-qazc9,sub-pixar073,j28ca
-dc5fm,sub-pixar074,xwvna
-b4xfs,sub-pixar077,2fmqn
-fbaw5,sub-pixar080,yp6a5
-z3ah7,sub-pixar089,ypk98
-6syqa,sub-pixar090,hvj9r
-nm6at,sub-pixar097,7j94p
-kymu7,sub-pixar103,hps2d
-vfc72,sub-pixar109,3z4gc
-3nvkb,sub-pixar110,qktms
-56948,sub-pixar113,nke3a
-ef8wp,sub-pixar116,javzc
-em3su,sub-pixar117,yksae
-fbtak,sub-pixar121,xq9ns
-e4kmw,sub-pixar123,yjmrf
-j7zgf,sub-pixar124,szk5u
-u2f4d,sub-pixar125,3kfrh
-jpfr9,sub-pixar126,ajsd3
-tu9p7,sub-pixar127,c7psm
-gj9ut,sub-pixar128,spjqy
-sxf5a,sub-pixar129,t85b6
-z8nwd,sub-pixar130,589me
-xvyd4,sub-pixar131,6q5gt
-kcw3b,sub-pixar132,anq43
-84fk2,sub-pixar133,y7gub
-b2skw,sub-pixar134,9264h
-zxhyn,sub-pixar135,7b3at
-5vzxf,sub-pixar136,nxh6v
-mjp3n,sub-pixar137,5pu7g
-739gk,sub-pixar138,uq76n
-g7vz8,sub-pixar139,gftju
-kunmx,sub-pixar140,twj6r
-vbcma,sub-pixar141,3vxzr
-j2q3k,sub-pixar142,93wfs
-s9ru8,sub-pixar143,h98fy
-bxqvg,sub-pixar144,ej7w2
-ug3s9,sub-pixar145,jtyw4
-6pm7e,sub-pixar146,k497t
-suqv3,sub-pixar147,8te3p
-w23s6,sub-pixar148,nha7y
-h9fsy,sub-pixar149,jebsq
-kw4ut,sub-pixar150,3jtwu
-dbw8y,sub-pixar151,gwns9
-rdchv,sub-pixar152,mhjrk
-sfkuh,sub-pixar153,btpv9
-7xzdq,sub-pixar154,yctrz
-b95rn,sub-pixar155,ut6nj
+participant_id,key_r,key_b
+sub-pixar001,5c8ff37da743a90018606df1,5c8ff37c2286e80019c3c102
```

Comment:
Do you know if we can automatically grab these keys ?

Reply:
Yes --  @rmarkello showed me how to use the OSF API :)

It looked like this:
```python
url = 'https://api.osf.io/v2/nodes/5hju4/files/osfstorage/?filter[name][contains]=confounds_regressors.tsv'
guids = []
while True:
    resp = requests.get(url)
    resp.raise_for_status()
    data = json.loads(resp.content)
    for i in data['data']:
        sub = re.search(r'sub-(\S+)_task', i['attributes']['name']).group(1)
        guids.append((sub, i['id']))

    url = data['links']['next']
    if url is None:
        break

```

Reply:
Sounds great. Thanks a lot!